sdk/servicebus/service-bus/samples/javascript/receiveMessagesStreaming.js

```diff
@@ -23,31 +30,47 @@ async function main() {
   // If receiving from a Subscription, use `createSubscriptionClient` instead of `createQueueClient`
   const queueClient = sbClient.createQueueClient(queueName);
 
-  // To receive messages from sessions, use getSessionReceiver instead of getReceiver or look at
+  // To receive messages from sessions, use createSessionReceiver instead of createReceiver or look at
   // the sample in sessions.ts file
-  const receiver = queueClient.createReceiver(ReceiveMode.peekLock);
-
-  const onMessageHandler = async (brokeredMessage) => {
-    console.log(`Received message: ${brokeredMessage.body}`);
-    await brokeredMessage.complete();
-  };
-  const onErrorHandler = (err) => {
-    console.log("Error occurred: ", err);
-  };
-
-  try {
-    receiver.registerMessageHandler(onMessageHandler, onErrorHandler, {
-      autoComplete: false
+
+  // controls whether we continue to recover from fatal Receiver failures.
+  let enableReceiverRecovery = true;
+
+  do {
+    const receiver = queueClient.createReceiver(ReceiveMode.peekLock);
+
+    const receiverPromise = new Promise((resolve, _reject) => {
+      const onMessageHandler = async (brokeredMessage) => {
+        console.log(`Received message: ${brokeredMessage.body}`);
+        await brokeredMessage.complete();
+      };
+
+      const onErrorHandler = (err) => {
+        if (err.retryable === false) {
+          console.log("Receiver will be recreated. A fatal error occurred:", err);
+          resolve();
```

Comment:
Instead of non retryable errors, shouldnt the sample retry when the error is retryable and the error is not from lock renewal or message settlement.


The case we saw in #9958 was that of a timeout error which AFAIK is retryable. This sample would not help that case.

Reply:
Yes, this is just the base. To "handle" the case from #9958 we need to modify it to basically include _any_ error that has .retryable set but I wasn't sure if we should do that in the one we're generally offering.

Reply:
Resolving with PR #10058